testpar/t_pflush2.c

```diff
@@ -213,10 +217,18 @@ main(int argc, char *argv[])
     h5_clean_files(&FILENAME[0], fapl_id1);
     h5_clean_files(&FILENAME[1], fapl_id2);
 
+    if (data_g) {
+        HDfree(data_g);
+        data_g = NULL;
+    }
+
     MPI_Finalize();
 
     HDexit(EXIT_SUCCESS);
 
 error:
+    if (data_g)
+        HDfree(data_g);
```

Comment:
The check for data_g isn't necessary. You can call free() on a NULL pointer.

Reply:
I know; bad habit